### ISSUE TITLE
Setting enable.ssl.certificate.verification=false causes sasl.mechanisms=SCRAM-SHA-512 to fail

### ISSUE THREAD

**Issue by rajha-korithrien** [edenhill/kcat#363]
If I provide kafkacat with the CA cert to verify the brokers TLS certs I can use SASL `scram-sha-512` to authenticate:
```
kafkacat -b test-kafka-kafka-bootstrap:9093 -X ssl.ca.location=/etc/kafka/ca.crt -X security.protocol=SASL_SSL -X sasl.mechanisms=SCRAM-SHA-512 -X sasl.username="test-regular" -X sasl.password="redacted" -L
Metadata for all topics (from broker -1: sasl_ssl://mwp-test-kafka-kafka-bootstrap:9093/bootstrap):
 1 brokers:
  broker 0 at test-kafka-kafka-0.test-kafka-kafka-brokers.kafka-test.svc:9093 (controller)
 4 topics:
  topic "__strimzi_store_topic" with 1 partitions:
    partition 0, leader 0, replicas: 0, isrs: 0
...
...
```
If I use the same credentials for SASL, but ask for the broker TLS certs to be ignored by setting `enable.ssl.certificate.verification=false` the connection fails:
```
kafkacat -b test-kafka-kafka-bootstrap:9093 -X enable.ssl.certificate.verification=false -X security.protocol=SASL_SSL -X sasl.mechanisms=SCRAM-SHA-512 -X sasl.username="test-regular" -X sasl.password="redacted" -L
% ERROR: Failed to acquire metadata: Local: Broker transport failure
```

I expect it to work the same in both cases, ignoring the TLS cert of the broker should not impact the SASL/SCRAM-SHA-512 authentication step.

I also suspect it is a librdkafka thing and not a kcat thing, but I can't prove it.
Version details:
```
kafkacat - Apache Kafka producer and consumer tool
https://github.com/edenhill/kafkacat
Copyright (c) 2014-2019, Magnus Edenhill
Version 1.5.0 (JSON, Avro, librdkafka 1.1.0 builtin.features=gzip,snappy,ssl,sasl,regex,lz4,sasl_gssapi,sasl_plain,sasl_scram,plugins,zstd,sasl_oauthbearer)
```

Thanks for looking!

**Comment by edenhill** [comment]
1. Upgrade to the latest version of librdkafka (v1.8.2)
2. Reproduce with `-d broker,security,protocol` and provide the logs.

**Comment by rajha-korithrien** [comment]
My original submission was using the container image: `confluentinc/cp-kafkacat:7.0.1-3-ubi8`

I don't find a container image that provides librdkafka 1.8.2, but I do find one that provides `librdkafka 1.7.0` via `edenhill/kcat:1.7.0`

When using the container `edenhill/kcat:1.7.0` it works as expected.
When using the container `confluentinc/cp-kafkacat:7.0.1-3-ubi8` I get the previously described error.

I can't tell you if that is because of the differences in the base container, alpine vs. ubi8 which have very different libc and ssl implementations. Do you still want logs, or shall I close this?

Thanks!

**Comment by edenhill** [comment]
That sounds weird indeed, must be something with openssl in the ubi8 image.

I'll release a v1.8.2-bundled version of edenhill/kcat.

**Comment by edenhill** [comment]
Try `edenhill/kcat:1.7.1` docker image

**Comment by rajha-korithrien** [comment]
To ensure we are comparing across the same base container, I hand built a ubi8 based image with the `1.7.1` kcat branch, which uses `librdkafka 1.8.2`

It works as expected:
```
bash-4.4$ kcat -V
kcat - Apache Kafka producer and consumer tool
https://github.com/edenhill/kcat
Copyright (c) 2014-2021, Magnus Edenhill
Version 1.7.1 (JSON, Avro, Transactions, IncrementalAssign, JSONVerbatim, librdkafka 1.8.2 builtin.features=gzip,snappy,ssl,sasl,regex,lz4,sasl_gssapi,sasl_plain,sasl_scram,plugins,zstd,sasl_oauthbearer)

bash-4.4$ kcat -b test-kafka-kafka-bootstrap:9093 -X security.protocol=SASL_SSL -X enable.ssl.certificate.verification=false -X sasl.mechanisms=SCRAM-SHA-512 -X sasl.username="test-regular" -X sasl.password="redacted" -L
Metadata for all topics (from broker -1: sasl_ssl://test-kafka-kafka-bootstrap:9093/bootstrap):
 1 brokers:
  broker 0 at test-kafka-kafka-0.test-kafka-kafka-brokers.kafka-test.svc:9093 (controller)
 4 topics:
  topic "__strimzi_store_topic" with 1 partitions:
    partition 0, leader 0, replicas: 0, isrs: 0
  topic "testing" with 1 partitions:
    partition 0, leader 0, replicas: 0, isrs: 0
  topic "__strimzi-topic-operator-kstreams-topic-store-changelog" with 1 partitions:
    partition 0, leader 0, replicas: 0, isrs: 0
  topic "__consumer_offsets" with 50 partitions:
    partition 0, leader 0, replicas: 0, isrs: 0
```
This confirms 2 things:
1) The original issue of different behavior with `enable.ssl.certificate.verification=false` is not caused by the difference between alpine and ubi8
2) The original issue was resolved by changes made between `librdkafka 1.1.0` and `librdkafak 1.8.2`
3) Until the `confluentinc/cp-kafkacat` containers catch up to a version of `librdkafak` that doesn't behave this way, they will exhibit this issue.
